pkg/validations/upgradevalidations/cluster.go

```diff
@@ -28,8 +28,21 @@ func ValidateClusterObjectExists(ctx context.Context, k validations.KubectlClien
 
 func ValidateTaintsSupport(ctx context.Context, clusterSpec *cluster.Spec) error {
 	if !features.IsActive(features.TaintsSupport()) {
-		if len(clusterSpec.Cluster.Spec.ControlPlaneConfiguration.Taints) > 0 {
-			return fmt.Errorf("Taints feature is not enabled.")
+		if len(clusterSpec.Cluster.Spec.ControlPlaneConfiguration.Taints) > 0 ||
+			len(clusterSpec.Cluster.Spec.WorkerNodeGroupConfigurations[0].Taints) > 0 {
+			return fmt.Errorf("Taints feature is not enabled. Environment variable TAINTS_SUPPORT needs to be set to true.")
```

Comment:
Is the feature is not active and we notice taints, don't we want the error not show that it can be enabled? The feature flag is more for us developers than customer facing right?

Reply:
This is what we decided during control plane taints changes. I think this validation required if we want to feature gate this behavior and release. What do you suggest should happen instead, if this is not enabled and a user have Taints specified in the cluster spec.

Reply:
I was thinking it would error saying unknown field, but I guess this feature gate will be short lived and we won't have any documentation about it.